src/test/java/plugins/LdapPluginTest.java

```diff
@@ -240,6 +299,72 @@ public void resolve_email() {
     }
 
     /**
+     * Scenario: do not resolve email address
+     * Given I have a docker fixture "ldap"
+     * And Jenkins is using ldap as security realm and resolve email address is disabled
+     * When I login with user "jenkins" and password "root"
+     * Then I will be logged on as user "jenkins"
+     * And there will not be any resolved email address"
+     */
+    @Test
+    public void do_not_resolve_email() {
+        // Given
+        LdapDetails details = createDefaults(ldap.get());
+        details.setDisableLdapEmailResolver(true);
+        useLdapAsSecurityRealm(details);
+        // When
+        Login login = jenkins.login();
+        login.doLogin("jenkins", "root");
+        // Then
+        assertThat(jenkins, hasLoggedInUser("jenkins"));
+        User u = new User(jenkins, "jenkins");
+        assertThat(u.mail(), nullValue());
+    }
+    
+   /**
+   * Scenario: do not resolve email address
+   * Given I have a docker fixture "ldap"
+   * And Jenkins is using ldap as security realm and cache is enabled
+   * When I login with user "jenkins" and password "root"
+   * Then I will be logged on as user "jenkins"
+   * Nothing much can be tested here apart from being able to use the security realm with the option activated
+   */
+    @Test
+    public void enable_cache() throws IOException {
+        // Given
+        LdapDetails details = createDefaults(ldap.get());
+        details.setEnableCache(true);
+        useLdapAsSecurityRealm(details);
+        // When
+        Login login = jenkins.login();
+        login.doLogin("jenkins", "root");
+        // Then
+        assertThat(jenkins, hasLoggedInUser("jenkins"));
+    }
+    
+    /**
+     * Scenario: can use environment variables
+     * Given I have a docker fixture "ldap"
+     * And Jenkins is using ldap as security realm and the java.naming.ldap.typesOnly env var is se to true
+     * When I login with user "jenkins" and password "root"
+     * Then I will not be logged on as user "jenkins" because the values for the LDAP attributes
+     * have not been sent as defined in the environment variable.
+     */
+    @Test
+    public void use_environment_varibales() {
+        // Given
+        LdapDetails details = createDefaultsWithoutManagerCred(ldap.get());
+        details.addEnvironmentVariable(new LdapEnvironmentVariable("java.naming.ldap.typesOnly", "true"));
+        useLdapAsSecurityRealm(details);
+        // When
+        Login login = jenkins.login();
+        login.doLogin("jenkins", "root");
+        // Then
+        assertThat(jenkins, not(hasLoggedInUser("jenkins")));
+        assertThat(getElement(by.tagName("pre")).getText(), Matchers.containsString("NoSuchElementException"));
```

Comment:
Yes... may be in case the exception thrown is different. In any case, what I wanted to test is that the environment variables are actually being used, and from [documentation](http://docs.oracle.com/javase/7/docs/technotes/guides/jndi/jndi-ldap.html) this one seemed like an easy one to test. Removed the exception assertion and just checked that the user was not logged in, that should be alright.